### PR TITLE
Implement confirmation check

### PR DESCRIPTION
**Pull request by FionaQY**
Currently, there is no check after a clear/delete command is given. Now, users are prompted with a message to ask them for confirmation before the command is executed.

Closes https://github.com/AY2425S1-CS2103T-T14-4/tp/issues/106

### CODE REVIEW

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 1]
```diff
@@ -27,12 +27,18 @@ public class LogicManager implements Logic {
     public static final String FILE_OPS_PERMISSION_ERROR_FORMAT =
             "Could not save data to file %s due to insufficient permissions to write to the file or the folder.";
 
+    public static final String MESSAGE_COMMAND_CANCELLED = "Command has been cancelled.";
```
Comment (ZShunRen) [inline src/main/java/seedu/address/logic/LogicManager.java:30]: Perhaps this string can be located in `Messages.java` for better standardisation

File: src/main/java/seedu/address/logic/commands/ClearCommand.java [hunk 2]
```diff
@@ -12,12 +12,24 @@ public class ClearCommand extends Command {
...
     }
+
+    @Override
+    public CommandResult execute(Model model, Boolean confirmationReceived) {
+        if (confirmationReceived.equals(requiresConfirmation)) {
+            return this.execute(model);
+        }
+        return new CommandResult(MESSAGE_CLEAR_CONFIRMATION, false, false, true);
+    }
+
```
Comment (ZShunRen) [inline src/main/java/seedu/address/logic/commands/ClearCommand.java:34]: Perhaps, just leave it as `confirmationReceived.equals(true)`

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 3]
```diff
@@ -84,4 +84,15 @@ public Command parseCommand(String userInput) throws ParseException {
...
+     * Parses user input into command for execution.
+     *
+     * @param userInput full user input string
+     * @return boolean depending on whether the input is yes/y
+     */
+    public boolean parseConfirmation(String userInput) {
+        userInput = userInput.replace("\n", "").toLowerCase().trim();
+        return userInput.equals("y") || userInput.equals("yes");
+    }
+
```
Comment (ZShunRen) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:97]: perhaps use `userInput.toLowerCase().equals("yes")`

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 4]
```diff
@@ -60,9 +61,16 @@ public void execute_invalidCommandFormat_throwsParseException() {
...
+            throws CommandException, ParseException {
         String deleteCommand = "delete 9";
-        assertCommandException(deleteCommand, MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        assertCommandSuccess(deleteCommand, MESSAGE_DELETE_CONFIRMATION, model);
+    }
+
+    @Test
+    public void execute_commandExecutionError_throwsCommandException() throws CommandException, ParseException {
+        String[] deleteCommand = {"delete 9", "y"};
+        assertDeleteCommandFailure(CommandException.class, MESSAGE_INVALID_PERSON_DISPLAYED_INDEX, deleteCommand);
```
Comment (ZShunRen) [inline src/test/java/seedu/address/logic/LogicManagerTest.java:73]: nice work with these tests, but perhaps you can try asserting that if a non `y` or `yes` value is provided, then the cancellation message is shown?

**Review by ZShunRen — APPROVED** [review]
Overall LGTM, just a few nitpicks. Great work!

File: src/main/java/seedu/address/logic/commands/ClearCommand.java (hunk 2 above)
Comment (FionaQY) [reply src/main/java/seedu/address/logic/commands/ClearCommand.java:34]: I mean if requiresConfirmation is false, then this.execute(model) should happen as well

File: src/test/java/seedu/address/logic/LogicManagerTest.java (hunk 4 above)
Comment (FionaQY) [reply src/test/java/seedu/address/logic/LogicManagerTest.java:73]: Good idea.

File: src/main/java/seedu/address/logic/commands/ClearCommand.java (hunk 2 above)
Comment (ZShunRen) [reply src/main/java/seedu/address/logic/commands/ClearCommand.java:34]: i see, then no issues with this.

**Review by itsme-zeix — APPROVED** [review]
Great work, this would be a super helpful feature! 

Could we also update the UG to reflect these changes?

**Pull request merged**